pyomo/gdp/plugins/bigm.py

```diff
@@ -150,15 +156,29 @@ def _get_bigm_suffix_list(self, block):
             block = block.parent_block()
         return suffix_list
 
+    def _get_bigm_arg_list(self, bigm_args, block):
+        # Gather what we know about blocks from args exactly once. We'll still
+        # check for constraints in the moment, but if that fails, we've
+        # preprocessed the time-consuming part of traversing up the tree.
+        arg_list = []
+        if bigm_args is None:
+            return arg_list
+        while block is not None:
+            if block in bigm_args:
+                arg_list.append({block: bigm_args[block]})
+            block = block.parent_block()
+        return arg_list
 
     def _apply_to(self, instance, **kwds):
         assert not NAME_BUFFER
+        assert not used_args
```

Comment:
Why is this a global?  How hard would it be to make this a local variable (or at least a variable attached to this transformation instance)?

Reply:
Laziness... It's easy, you'd just have to pass it all the way down all the function calls because it doesn't get used until you are getting an M value while transforming a constraint. So we could either do that or attach it to the transformation instance. I'll attach it to the instance unless you have objections?